src/main/java/org/sonarlint/intellij/actions/SonarAnalyzeChangedFilesAction.java

```diff
@@ -53,17 +55,38 @@
     }
 
     SonarLintSubmitter submitter = SonarLintUtils.get(project, SonarLintSubmitter.class);
-    ChangedFilesIssues changedFilesIssues = SonarLintUtils.get(project, ChangedFilesIssues.class);
     ChangeListManager changeListManager = ChangeListManager.getInstance(project);
 
+    AnalysisCallback callback = new ShowIssuesCallable(project);
     List<VirtualFile> affectedFiles = changeListManager.getAffectedFiles();
-    submitter.submitFiles(affectedFiles, TriggerType.ACTION, false);
+    submitter.submitFiles(affectedFiles, TriggerType.ACTION, callback, false);
+  }
+
+  private class ShowIssuesCallable implements AnalysisCallback {
+    private final Project project;
+
+    private ShowIssuesCallable(Project project) {
+      this.project = project;
+    }
 
-    // TODO: do something in case of error?
-    IssueManager issueManager = SonarLintUtils.get(project, IssueManager.class);
-    Map<VirtualFile, Collection<LiveIssue>> map = affectedFiles.stream()
-      .collect(Collectors.toMap(Function.identity(), issueManager::getForFile));
+    @Override public void onSuccess(Map<VirtualFile, Collection<LiveIssue>> issues) {
+      ChangedFilesIssues changedFilesIssues = SonarLintUtils.get(project, ChangedFilesIssues.class);
+      changedFilesIssues.set(issues);
+      showChangedFilesTab(project);
+    }
+
+    @Override public void onError(Exception e) {
+      // do nothing
+    }
+  }
 
-    changedFilesIssues.set(map);
+  private void showChangedFilesTab(Project project) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make "showChangedFilesTab" a "static" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this method into "ShowIssuesCallable". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3398)